rar.go

```diff
@@ -37,6 +38,23 @@ func Unrar(source, destination string) error {
 		} else if err != nil {
 			return err
 		}
+		
+		pathComponents := strings.Split(header.Name, "/")
```

Comment:
You could do `filepath.Dir(header.Name)` and avoid that `if pi == len(pathComponents) - 1` check below.